src/stdlib_stats.fypp

```diff
@@ -305,11 +306,65 @@ module stdlib_stats
   end interface mean
 
 
+  interface median
+    !! version: experimental
+    !!
+    !! Median of array elements
+    !! ([Specification](../page/specs/stdlib_stats.html#median-median-of-array-elements))
+    #:for k1, t1, o1 in IR_KINDS_TYPES_OUTPUT
+      #:for rank in RANKS
+        #:set RName = rname("median_all",rank, t1, k1, o1)
+        module function ${RName}$ (x, mask) result(res)
+          ${t1}$, intent(in) :: x${ranksuffix(rank)}$
+          logical, intent(in), optional :: mask
```

Comment:
This is just a question. I couldn't really grasp the purpose of median with the scalar optional mask? The false case will always give NaN, and the true case is the same as having no mask, right?

Is the use case supposed to be something like:

```
x = [...]
y = median(x, all(x > 0))
```

Reply:
Same, I'm confused about this as well.

Reply:
This is to be in agreement with the API of the intrinsic `sum`, and the already implemented `mean`, `var`,...
Such API will indeed cover cases as proposed by @ivan-pi

Reply:
Yes, this is indeed the case for [SUM](https://gcc.gnu.org/onlinedocs/gfortran/SUM.html). It would be interesting to learn the output of `sum(array,.false.)`. Is it also NaN?

Reply:
`sum(array, .false.)` will return `0`, as following the standard.

The functions `mean`, `var`, .... will return `NaN`, because of `mean(array, .false.) = sum(array, .false.) / 0`

For `median`, if  all elements correspond to `.false.`, then there are no elements for `median`. Hence, it makes sense to return `NaN` IMO.